### PR TITLE
Itay and noam

### PR DESCRIPTION
**Pull request by noamTzuber**

### CODE REVIEW

File: .idea/GifParser.iml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <content url="file://$MODULE_DIR$">
```
Comment (NeviF) [inline .idea/GifParser.iml]: remove this file from git

File: .idea/misc.xml [hunk 2]
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (NeviF) [inline .idea/misc.xml]: remove this file from git

File: decoder.py [hunk 3]
```diff
@@ -80,7 +119,7 @@ def decode_logical_screen_descriptor(gif_stream: typing.BinaryIO, gif_object: Gi
     if global_color_table_exist:
         gif_object.global_color_table_size = 3 * pow(2, bits_to_int(packed_bits, 5, 7) + 1)
```
Comment (NeviF) [inline decoder.py:120]: replace all the magic numbers , this line is unreadable.

File: decoder.py [hunk 4]
```diff
@@ -137,39 +173,66 @@ def decode_image_descriptor(gif_stream: typing.BinaryIO, gif_object: Gif) -> Non
...
         bytes_image_data += decode_lzw(compressed_sub_block, math.pow(2, lzw_minimum_code_size))
 
-    # local_color_table = gif_object.LCTs[-1]
+    if current_image.local_color_table_flag == 1:
+        local_color_table = gif_object.local_color_tables[-1]
+    else:
+        local_color_table = gif_object.global_color_table
+
+    pos = 0
+    for i in range(int(len(bytes_image_data) / index_length)):
```
Comment (NeviF) [inline decoder.py]: use pos as the for index and add to range the step size

File: main.py [hunk 5]
```diff
@@ -8,16 +8,17 @@ def main():
 
 
 if __name__ == '__main__':
-    # main()
-    gif = gif_objects.Gif()
-    with open("example.gif", "rb") as file:
+    main()
+    i =0
+    # gif = gif_objects.Gif()
```
Comment (NeviF) [inline main.py]: remove redundant comments

File: decoder.py (hunk 3 above)
Comment (noamTzuber) [reply decoder.py:120]: @NoamCohen48

Comment (NoamCohen48) [reply decoder.py:120]: i will change at my branch

**Review by NeviF — APPROVED** [review]

**Pull request merged**